postgres-native-sqldelight-driver/src/commonTest/kotlin/app/softwork/sqldelight/postgresdriver/PostgresNativeDriverTest.kt

```diff
@@ -208,4 +214,107 @@ class PostgresNativeDriverTest {
         val results = driver.copy("1\n2\n")
         assertEquals(2, results)
     }
+
+    @Test
+    fun remoteListenerTest() = runTest {
+        val other = PostgresNativeDriver(
+            host = "localhost",
+            port = 5432,
+            user = "postgres",
+            database = "postgres",
+            password = "password",
+            listenerSupport = ListenerSupport.Remote(backgroundScope)
+        )
+
+        val driver = PostgresNativeDriver(
+            host = "localhost",
+            port = 5432,
+            user = "postgres",
+            database = "postgres",
+            password = "password",
+            listenerSupport = ListenerSupport.Remote(backgroundScope)
+        )
+
+        val results = MutableStateFlow(0)
+        val listener = object : Query.Listener {
+            override fun queryResultsChanged() {
+                results.update { it + 1 }
+            }
+        }
+        driver.addListener(listener, arrayOf("foo", "bar"))
+        withContext(Dispatchers.Default) {
+            val setupSocket = 2.seconds
+            delay(setupSocket)
```

Comment:
Should these tests use `runBlocking { }` instead of `runTest { }` 🤔 I assume this `withContext` is in-place to avoid the test behavior for `delay`?

Reply:
Okay, you can't close a hot flow. It will be automatically cleaned up by garbage collector. The only thing I can do is canceling the scope and make all (except for testing) references internal to remove all references.